vllm/v1/request.py

```diff
@@ -65,6 +77,9 @@ def __init__(
         self.output_token_ids = ConstantList(self._output_token_ids)
         self.all_token_ids = ConstantList(self._all_token_ids)
 
+        # Grammar fields, including the grammar object and the bitmask
+        self._grammar: Optional[Union[Future[Grammar], Grammar]] = None
```

Comment:
🤔 Can we do something like `LoRARequest` to hide all the grammar-related details from the `Request` class? I think we should keep the `Request` class as simple and minimal as possible since it's used in many system components and it's what people will be faced with when reading the V1 code. I feel many of the new methods added in this PR are too specific to Grammar.

Reply:
Yeah I think this makes sense, I will do a quick refactor to address this.